[Tang Yukun] iP

gandwarf opened the pull request:
Alpha

> _Productivity is never an accident. It is always the result of a commitment to excellence." – Paul J. Meyer_[(source)](https://monday.com/blog/productivity/productivity-quotes/#:~:text=It%20is%20always%20the%20result%20of%20a%20commitment,productive%20isn%E2%80%99t%20something%20that%20you%20are%20born%20with.)

Feeling overwhelmed by millions of works? Let Alpha help you deal with it!
Equipped with 
- text-based📄
- No entry barrier🎯
- super fast⚡⚡
Alpha will do its best

Let's get start:

1. download the newest release
2. run it using `java -jar alpha.java`
3. add all your tasks
4. let it git rid of all the trouble for you

Features:

- [x] managing tasks 
- [x] reminding deadlines
- [x] arranging events
- [ ] more to come

```public class Main {
    public static void main(String[] args) {
        // says hi.
        System.out.println("Hello, world!!");
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

gandwarf committed:
Refactor: Replace Duke.java with Alpha.java

gandwarf committed:
[200~Add Task class and improve task handling in Alpha

gandwarf committed:
Modify Alpha.java and Task.java

gandwarf committed:
Add ToDo.java, Event.java adn Deadline.java Update Task.java and Alpha.java

gandwarf committed:
Update Alpha.java

gandwarf committed:
Upate Alpha.java to add delete method

gandwarf committed:
Level-2 to Level-6(reupload)

gandwarf committed:
Update EXPECTED.TXT and INPUT.TXT

gandwarf committed:
Add save features to Alpha

gandwarf committed:
Merge branch-Level-7 into master

gandwarf committed:
Update date feature to Alpha

gandwarf committed:
Merge branch-Level-8 into master

gandwarf committed:
Add increments

gandwarf committed:
Merge branch 'add-gradle-support'

gandwarf committed:
Implements all increments in 4

gandwarf committed:
Add JavaDoc

gandwarf committed:
Adjust to the java coding style

gandwarf committed:
Add find method to alpha

gandwarf committed:
Resolve merge conflict in Alpha.java

gandwarf committed:
Merge branch 'branch-Level-9'

blobfish465 commented on src/main/java/alpha/Alpha.java:
```diff
@@ -0,0 +1,112 @@
+            String[] words = input.split("\\s+", 2);
+            try {
+                switch (words[0]) {
+                    case "list" -> taskList.showList();
```
Shouldn't the switch and case statements have the same indentation?

gandwarf replied:
![image](https://github.com/user-attachments/assets/7388bb93-17ca-4b1c-ad57-3c8755e8a786)
lambda version needs indentation

blobfish465 commented on src/main/java/alpha/task/TaskList.java:
```diff
@@ -0,0 +1,155 @@
+ * Represents a list of tasks. Provides methods to add, remove, and modify tasks.
+ */
+public class TaskList {
+    private final ArrayList<Task> tasks;
```
Great job following the correct convention for tasks as a plural noun for collections. No changes needed.

blobfish465 commented on src/main/java/alpha/task/Task.java:
```diff
@@ -0,0 +1,71 @@
+    /**
+     * Indicates whether this task is marked as done.
+     */
+    protected boolean done;
```
Should this be renamed to isDone to follow boolean naming  conventions?

blobfish465 commented on src/main/java/alpha/Alpha.java:
```diff
@@ -0,0 +1,112 @@
+package alpha;
```
Great job following the package naming convention of all lowercase. No Changes needed.

blobfish465 commented on src/main/java/alpha/Storage.java:
```diff
@@ -0,0 +1,142 @@
+    /**
+     * Represents the current working directory of the user.
+     */
+    private static final Path HOME = Paths.get("").toAbsolutePath();
```
Good job, constant names must indeed be all uppercase, using underscore to seperate words. No changes needed.

blobfish465 reviewed (commented):
Overall, i there were no major issues in terms of naming code quality and naming guidelines. There is only some minor changes to consider such as boolean variable naming "done". I like the way you applied OOP into this project, splitting into smaller classes. Great job!

gandwarf commented:
> _ No description provided. _

gandwarf committed:
Modify some configuration in build.gradle

gandwarf committed:
A-CheckStyle

gandwarf committed:
Add a GUI

gandwarf committed:
S

gandwarf committed:
Implement Level-10 GUI

gandwarf committed:
Implement Varargs

gandwarf committed:
Use Java assert to document important assumptions

Parts of the code rely on assumptions about states and arguments but lack explicit checks. Without runtime validation, these assumptions can be overlooked, risking subtle bugs.

This commit adds Java assert statements to relevant methods in classes such as Alpha, Main, and TaskList to highlight these assumptions. Doing so clarifies developer intent and helps detect issues early when assertions are enabled.

Let's also add new images (ikaros.png, yui.png) for the main window’s UI enhancements and update build.gradle to ensure they are packaged properly.

gandwarf committed:
Merge pull request #1 from gandwarf/branch-A-Assertions

Branch a assertions

gandwarf committed:
A-CodeQuality: refine Alpha structure and introduce Command class

Alpha handles multiple responsibilities, making code maintenance harder. The data file (alpha.txt) also lacks clarity about how commands are processed, causing confusion when new features are introduced.

A clearer separation of concerns is needed to improve code quality and simplify future enhancements.

We introduce a Command class to encapsulate command handling while updating Alpha to delegate command-related tasks. The alpha.txt file is also refined to align with the new command architecture.

Following established code quality practices ensures the codebase remains maintainable. Extracting command logic into its own class clarifies each component's purpose and makes the system easier to extend.

gandwarf committed:
Merge branch 'master' into branch-A-CodeQuality

gandwarf committed:
A-CodeQuality: refine Alpha structure and introduce Command class

Alpha handles multiple responsibilities, making code maintenance harder. The data file (alpha.txt) also lacks clarity about how commands are processed, causing confusion when new features are introduced.

A clearer separation of concerns is needed to improve code quality and simplify future enhancements.

We introduce a Command class to encapsulate command handling while updating Alpha to delegate command-related tasks. The alpha.txt file is also refined to align with the new command architecture.

Following established code quality practices ensures the codebase remains maintainable. Extracting command logic into its own class clarifies each component's purpose and makes the system easier to extend.

gandwarf committed:
Merge pull request #2 from gandwarf/branch-A-CodeQuality

Branch a code quality

gandwarf committed:
Refactor Storage to use Java Streams

Storage operations previously used traditional loops and manual
collection handling, leading to verbosity and reduced readability.

Using Java Streams improves code clarity, reduces boilerplate,
and enhances maintainability.

Let's refactor Storage to leverage Streams for more concise
and expressive data processing.

gandwarf committed:
Merge pull request #3 from gandwarf/branch-A-Streams

Refactor Storage to use Java Streams

gandwarf committed:
C-DetectDuplicates: Handle duplicate task entries

TaskList previously allowed duplicate tasks to be added without any
checks. This could lead to cluttered task lists and redundant entries,
making task management inefficient.

A duplicate detection mechanism is now implemented to check if a task
already exists before adding it. When a duplicate is detected, the
system prompts the user for confirmation before proceeding with the
addition.

Let's modify TaskList to:
- Detect duplicate tasks using their string representation.
- Prompt the user through the UI to confirm whether to add the duplicate.
- Integrate this feature into the Command class for better usability.
- Update unit tests in TaskListTest to verify the new behavior.

This change ensures better task organization and prevents unintended
duplicate entries while still allowing users to manually override the
check if needed.

gandwarf committed:
Merge pull request #4 from gandwarf/branch-BCD-Extension

C-DetectDuplicates: Handle duplicate task entries

gandwarf committed:
Add files via upload

gandwarf committed:
Merge pull request #6 from gandwarf/Ui.png

Add files via upload

gandwarf committed:
Add files via upload

gandwarf committed:
Merge pull request #7 from gandwarf/DeadlineDemo

Add files via upload

gandwarf committed:
Add files via upload

gandwarf committed:
Merge pull request #8 from gandwarf/DuplicateDetection-Demo

Add files via upload

gandwarf committed:
Modify the userguide

gandwarf committed:
Merge pull request #9 from gandwarf/Published-UG

Published-Ug

gandwarf committed:
Modify README.mk

gandwarf committed:
Add a demo of ui

gandwarf committed:
Merge pull request #10 from gandwarf/Ui.png

Add a demo of ui

gandwarf committed:
Add files via upload

gandwarf committed:
Merge pull request #11 from gandwarf/Ui.png-1

Add files via upload

gandwarf committed:
Rename Ui.png.png to Ui.png

gandwarf committed:
some adjustion

gandwarf committed:
modify

gandwarf committed:
Delete A demo of Alpha.png

gandwarf committed:
Delete a demo of the execution of Duplicate Detection.png

gandwarf committed:
Delete a demo of the execution of deadline command.png

gandwarf committed:
Update README.md

gandwarf committed:
Update README.md

gandwarf committed:
Update README.md

gandwarf committed:
Add files via upload

gandwarf committed:
Update README.md

gandwarf committed:
Update README.md

gandwarf committed:
Update README.md

gandwarf committed:
Update README.md
